From 3ad44002b7387cbc4ffe0549f37b09d3b77b4333 Mon Sep 17 00:00:00 2001
From: Tamas K Lengyel <tamas@tklengyel.com>
Date: Fri, 2 Mar 2018 23:31:27 -0700
Subject: [PATCH 6/6] Don't measure with Authenticode

On TPM2 devices calculating the Authenticode hash needs to be performed
by the firmware. However, requesting measurements with PE_COFF_IMAGE flag
fails on certain firmwares (Dell). For consistency we fall back to measuring
the images as a whole without parsing the headers.
---
 shim.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/shim.c b/shim.c
index 0681c8c..386601d 100644
--- a/shim.c
+++ b/shim.c
@@ -1310,7 +1310,7 @@ static EFI_STATUS handle_image (void *data, unsigned int datasize,
 		return efi_status;
 
 	/* Measure the binary into the TPM */
-	efi_status = tpm_log_pe((EFI_PHYSICAL_ADDRESS)(UINTN)data, datasize, sha1hash, 4);
+	efi_status = tpm_log_pe((EFI_PHYSICAL_ADDRESS)(UINTN)data, datasize, NULL, 4);
 #ifdef REQUIRE_TPM
 	if (efi_status != EFI_SUCCESS) {
 		return efi_status;
@@ -1831,7 +1831,7 @@ EFI_STATUS shim_verify (void *buffer, UINT32 size)
 		goto done;
 
 	/* Measure the binary into the TPM */
-	status = tpm_log_pe((EFI_PHYSICAL_ADDRESS)(UINTN)buffer, size, sha1hash, 4);
+	status = tpm_log_pe((EFI_PHYSICAL_ADDRESS)(UINTN)buffer, size, NULL, 4);
 #ifdef REQUIRE_TPM
 	if (status != EFI_SUCCESS)
 		goto done;
-- 
2.1.4

